tools/sslsniff.py

```diff
@@ -48,41 +56,69 @@
                     help=argparse.SUPPRESS)
 parser.add_argument("--hexdump", action="store_true", dest="hexdump",
                     help="show data as hexdump instead of trying to decode it as UTF-8")
+parser.add_argument('--max-buffer-size', type=int, default=8192,
+                    help='Size of captured buffer')
 args = parser.parse_args()
 
 
 prog = """
 #include <linux/ptrace.h>
 #include <linux/sched.h>        /* For TASK_COMM_LEN */
 
+#define MAX_BUF_SIZE __MAX_BUF_SIZE__
+
 struct probe_SSL_data_t {
         u64 timestamp_ns;
         u32 pid;
-        char comm[TASK_COMM_LEN];
-        char v0[464];
+        u32 tid;
+        u32 uid;
         u32 len;
+        int buf_filled;
+        char comm[TASK_COMM_LEN];
+        u8 buf[MAX_BUF_SIZE];
 };
 
+#define BASE_EVENT_SIZE ((size_t)(&((struct probe_SSL_data_t*)0)->buf))
+#define EVENT_SIZE(X) (BASE_EVENT_SIZE + ((size_t)(X)))
+
+
+BPF_PERCPU_ARRAY(ssl_data, struct probe_SSL_data_t, 1);
 BPF_PERF_OUTPUT(perf_SSL_write);
 
 int probe_SSL_write(struct pt_regs *ctx, void *ssl, void *buf, int num) {
+        int ret;
+        u32 zero = 0;
         u64 pid_tgid = bpf_get_current_pid_tgid();
         u32 pid = pid_tgid >> 32;
+        u32 tid = pid_tgid;
+        u32 uid = bpf_get_current_uid_gid();
+
+        PID_FILTER
+        UID_FILTER
+        struct probe_SSL_data_t *data = ssl_data.lookup(&zero);
+        if (!data)
+            return 0;
+
+        data->timestamp_ns = bpf_ktime_get_ns();
+        data->pid = pid;
+        data->tid = tid;
+        data->uid = uid;
+        data->len = num;
+        data->buf_filled = 0;
+        bpf_get_current_comm(&data->comm, sizeof(data->comm));
+        u32 buf_copy_size = min((size_t)MAX_BUF_SIZE, (size_t)num);
+
+        if (buf != 0) {
+                ret = bpf_probe_read_user(data->buf, buf_copy_size, buf);
+        }
 
-        FILTER
-
-        struct probe_SSL_data_t __data = {0};
-        __data.timestamp_ns = bpf_ktime_get_ns();
-        __data.pid = pid;
-        __data.len = num;
-
-        bpf_get_current_comm(&__data.comm, sizeof(__data.comm));
-
-        if ( buf != 0) {
-                bpf_probe_read_user(&__data.v0, sizeof(__data.v0), buf);
+        if (!ret) {
+            data->buf_filled = 1;
+        } else {
+            buf_copy_size = 0;
         }
```

Comment:
The same as above, let remove '{}' since both then and else branch only have a single statement.